parlai/core/torch_agent.py

```diff
@@ -630,6 +647,18 @@ def init_optim(self, params, optim_states=None, saved_optim_type=None):
 
         optim_class = self.optim_opts()[opt['optimizer']]
         self.optimizer = optim_class(params, **kwargs)
+        if self.fp16:
+            try:
```

Comment:
since this code is repeated should we factor it into a helper method?

Reply:
very good idea. I'm much happier with the new verison